packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -324,16 +331,30 @@ if (__DEV__) {
   ReactStrictModeWarnings.flushLegacyContextWarning = () => {
     ((pendingLegacyContextWarning: any): FiberToFiberComponentsMap).forEach(
       (fiberArray: FiberArray, strictRoot) => {
-        const uniqueNames = new Set();
-        fiberArray.forEach(fiber => {
-          uniqueNames.add(getComponentName(fiber.type) || 'Component');
-          didWarnAboutLegacyContext.add(fiber.type);
-        });
-
-        const sortedNames = setToSortedString(uniqueNames);
-        const strictRootComponentStack = getStackByFiberInDevAndProd(
-          strictRoot,
-        );
+        const fibers = fiberArray.length;
+        let componentNames = '';
+
+        fiberArray
+          .sort((a, b) => {
+            const aCount = legacyContextCounts.get(a.type) || 0;
+            const bCount = legacyContextCounts.get(b.type) || 0;
+            return bCount - aCount;
+          })
```

Comment:
We don't actually need to sort the entire array by count, do we? We could just do a single pass through and find the highest count.

And we could build up the component names string and populate the `didWarnAboutLegacyContext` map in the same pass we compared count.

Reply:
Oh, I see. We were sorting by name before, and now we're showing the component names sorted by their count. I guess that makes sense then. 👍